distributed/platform.py

```diff
@@ -0,0 +1,70 @@
+import os
+import sys
+
+import psutil
+
+__all__ = ("memory_limit", "cpu_count", "PLATFORM_MEMORY_LIMIT", "PLATFORM_CPU_COUNT")
+
+
+def memory_limit():
+    """Get the memory limit (in bytes) for this system.
+
+    Takes the minimum value from the following locations:
+
+    - Total system host memory
+    - Cgroups limit (if set)
+    - RSS rlimit (if set)
+    """
+    limit = psutil.virtual_memory().total
+
+    # Check cgroups if available
+    if sys.platform == "linux":
+        try:
+            with open("/sys/fs/cgroup/memory/memory.limit_in_bytes") as f:
+                cgroups_limit = int(f.read())
+            if cgroups_limit > 0:
+                limit = min(limit, cgroups_limit)
+        except Exception:
+            pass
+
+    # Check rlimit if available
+    try:
+        import resource
+
+        hard_limit = resource.getrlimit(resource.RLIMIT_RSS)[1]
+        if hard_limit > 0:
+            limit = min(limit, hard_limit)
+    except (ImportError, OSError):
+        pass
+
+    return limit
+
+
+def cpu_count():
+    """Get the available CPU count for this system.
+
+    Takes the minimum value from the following locations:
+
+    - Total system cpus available on the host.
+    - Cgroups limit (if set)
+    """
+    count = os.cpu_count()
+
+    # Check cgroups if available
+    if sys.platform == "linux":
+        try:
+            with open("/sys/fs/cgroup/cpuacct,cpu/cpu.cfs_quota_us") as f:
+                quota = int(f.read())
+            with open("/sys/fs/cgroup/cpuacct,cpu/cpu.cfs_period_us") as f:
+                period = int(f.read())
+            cgroups_count = int(quota / period)
+            if cgroups_count > 0:
+                count = min(count, cgroups_count)
+        except Exception:
+            pass
+
+    return count
+
+
+PLATFORM_MEMORY_LIMIT = memory_limit()
+PLATFORM_CPU_COUNT = cpu_count()
```

Comment:
Thoughts on removing the `PLATFORM_` prefix here?  If we wanted the name then we could use Python namespaces like 

```python
from dask.distributed import platform

platform.CPU_COUNT
```

Reply:
Yeah, I thought about that. The issue here is that `platform` is already a python module. Happy to go with a different prefix and rename the module, naming is hard.

Reply:
Other potential module names (in the module scope of "query things about the host platform")):

- `system`
- `osutils`
- `os_utils`
- `sysutils`

Reply:
As I was writing "better names", I realized they were all taken or unclear. I think in general I would prefer to leave the current names, even if they feel less Pythonic/intuitive, IMHO clarity makes a big case here.

Reply:
I think that `platform.CPU_COUNT` is as clear as `PLATFORM_CPU_COUNT` or `system.CPU_COUNT` is as clear at `SYSTEM_CPU_COUNT`.  I'm slightly in favor of `system.CPU_COUNT`.

Reply:
My only issue with importing as `platform` was that it matches a stdlib module name. `system` does not, so I'm happy to switch to that and drop the prefixes.

Reply:
That would be my preference if that's ok with you.